tests/clims/models/test_substance.py

```diff
@@ -107,6 +107,63 @@ def do_asserts(substance):
         fresh_substance = Substance.objects.get(name=substance.name)
         do_asserts(self.app.substances.to_wrapper(fresh_substance))
 
+    def test_create_combined_sample__from_two_childs__origins_of_combined_are_original_samples(self):
+        # Arrange
+        props = dict(preciousness='*o*', color='red')
+        sample1 = self.create_gemstone(**props)
+        sample2 = self.create_gemstone(**props)
+        child1 = sample1.create_child()
+        child2 = sample2.create_child()
+
+        # Act
+        combined = GemstoneSample(
+            name='combined1', organization=self.organization,
+            parents=[child1, child2])
+        combined.save()
+
+        # Assert
+        assert len(combined.origins) == 2
+        assert set(combined.origins) == {sample1.id, sample2.id}
+
+    def test_set_property_for_combined_sample__fetched_object_from_db_ok(self):
+        # Arrange
+        props = dict(preciousness='*o*', color='red')
+        substance1 = self.create_gemstone(**props)
+        substance2 = self.create_gemstone(**props)
+
+        # Act
+        combined = GemstoneSample(
+            name='combined1', organization=self.organization,
+            parents=[substance1, substance2])
+        combined.preciousness = 'xxx'
+        combined.save()
+
+        # Assert
+        fetched = self.app.substances.get(name='combined1')
+
+        assert 'xxx' == fetched.preciousness
+        assert 2 == len(fetched.parents)
+
+    def test_depth__with_multiple_parents_with_different_depths__new_depth_based_on_highest(self):
+        # I just picked a policy for this, I had to choose something
+        # Arrange
+        props = dict(preciousness='*o*', color='red')
+        substance1 = self.create_gemstone(**props)
+        substance1._archetype.depth = 4
```

Comment:
It would be good to expose the depth on the SubstanceBase, but I think for now we can wait with it, since it's unclear if it will be called depth.

Reply:
Maybe we could leave this test as it is? I think this test is informative to show how depth works, and what is the current status of depth concept. When depth is renamed, it would be easy to adjust this test?

Reply:
Absolutely :)